src/screens/Profile/ProfileScreen.tsx

```diff
@@ -0,0 +1,239 @@
+import { MainStackParams, RootStackParams } from '../../Navigator';
+import { useAppDispatch, useAppSelector } from '../../hooks';
+import { TableCell } from '../../components/TableCell';
+import { spacing } from '../../util/spacing';
+import { accountLogout } from '../../store/thunks/accountLogout';
+import { colors } from '../../util/colors';
+import Avatar from '../../components/Avatar/Avatar';
+import { NavIconButton } from '../../components/NavIconButton';
+import { selectMailBoxAddress } from '../../store/selectors/email';
+import { Input } from '../../components/Input';
+import { accountUpdate } from '../../store/thunks/account';
+import styles from './styles';
+
+export type ProfileScreenProps = CompositeScreenProps<
+  NativeStackScreenProps<MainStackParams, 'profile'>,
+  NativeStackScreenProps<RootStackParams>
+>;
+
+export const ProfileScreen = ({ navigation }: ProfileScreenProps) => {
+  const dispatch = useAppDispatch();
+  const [displayName, setDisplayName] = useState<string>('');
+  const [avatar, setAvatar] = useState<string>('');
+  const [isEditing, setIsEditing] = useState<boolean>(false);
+  const [isImagePreparing, setIsImagePreparing] = useState<boolean>(false);
+  const inputRefVerify = useRef<TextInput>(null);
+
+  const account = useAppSelector(state => state.account);
+  const mailboxAddress = useAppSelector(selectMailBoxAddress);
+  const accountId = account.loginAccount?.accountId;
+  const userPlanText = `${account.stats.plan} Member`;
+
+  useEffect(() => {
+    setDisplayName(account.loginAccount?.displayName || '');
+    setAvatar(account.loginAccount?.avatar || '');
+  }, [account.loginAccount]);
+
+  useEffect(() => {
+    navigation.setOptions({
+      headerLeft: () => (
+        <NavIconButton
+          icon={{ name: 'menu-outline' }}
+          onPress={() => navigation.openDrawer()}
+        />
+      ),
+      headerRight: () => (
+        <View>
+          {!isEditing && (
+            <NavIconButton
+              icon={{
+                name: 'create-outline',
+              }}
+              onPress={() => setIsEditing(prev => !prev)}
+              padRight
+            />
+          )}
+        </View>
+      ),
+    });
+  }, [navigation, isEditing]);
+
+  const onLogout = () => {
+    dispatch(accountLogout());
+  };
+
+  const updateProfile = () => {
+    if (accountId) {
+      dispatch(
+        accountUpdate({
+          accountId: accountId,
+          displayName: displayName,
+          avatar: avatar,
+        }),
+      );
+    } else {
+      Alert.alert('Warning', 'An error occurred. Please try again later.');
+    }
+    setIsEditing(false);
+  };
+
+  const startProfileEdit = () => {
+    setIsEditing(true);
+  };
+
+  const selectImage = async () => {
+    setIsImagePreparing(true);
+    const result = await launchImageLibrary({
+      includeBase64: true,
+      mediaType: 'photo',
+    });
+    if (!result.didCancel) {
```

Comment:
What will happen if you will get an error here, like permission denied?

Reply:
error messages added 👍